x/auth/simulation/decoder/store_test.go

```diff
@@ -0,0 +1,63 @@
+package decoder
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/tendermint/tendermint/crypto/ed25519"
+	cmn "github.com/tendermint/tendermint/libs/common"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+
+	"github.com/cosmos/cosmos-sdk/x/auth/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+var (
+	delPk1    = ed25519.GenPrivKey().PubKey()
+	delAddr1  = sdk.AccAddress(delPk1.Address())
+	valAddr1  = sdk.ValAddress(delPk1.Address())
+	consAddr1 = sdk.ConsAddress(delPk1.Address().Bytes())
+)
+
+func makeTestCodec() (cdc *codec.Codec) {
+	cdc = codec.New()
+	sdk.RegisterCodec(cdc)
+	codec.RegisterCrypto(cdc)
+	types.RegisterCodec(cdc)
+	return
+}
+
+func TestDecodeStore(t *testing.T) {
+	cdc := makeTestCodec()
+	acc := types.NewBaseAccountWithAddress(delAddr1)
+	globalAccNumber := uint64(10)
+
+	kvPairs := cmn.KVPairs{
+		cmn.KVPair{Key: types.AddressStoreKey(delAddr1), Value: cdc.MustMarshalBinaryBare(acc)},
+		cmn.KVPair{Key: types.GlobalAccountNumberKey, Value: cdc.MustMarshalBinaryLengthPrefixed(globalAccNumber)},
+		cmn.KVPair{Key: []byte{0x99}, Value: []byte{0x99}},
+	}
+	tests := []struct {
+		name        string
+		expectedLog string
+	}{
+		{"Minter", fmt.Sprintf("%v\n%v", acc, acc)},
+		{"GlobalAccNumber", fmt.Sprintf("GlobalAccNumberA: %d\nGlobalAccNumberB: %d", globalAccNumber, globalAccNumber)},
+		{"other", ""},
+	}
+
+	for i, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			switch i {
+			case len(tests) - 1:
```

Comment:
what about using `expectedPanic` field? No need to change it, I am just curious for which method is preferred if I write table tests in the future

Reply:
either is fine tbh. I don't have a preference either